corehq/apps/saved_reports/tasks.py

```diff
@@ -174,12 +184,15 @@ def send_email_report(self, recipient_emails, domain, report_slug, report_type,
         if getattr(er, 'smtp_code', None) in LARGE_FILE_SIZE_ERROR_CODES or type(er) == ESError:
             # If the email doesn't work because it is too large to fit in the HTML body,
             # send it as an excel attachment.
+            request_params = json_request(request_data['GET'])
+            request_params['startdate'] = request_data['startdate']
+            request_params['enddate'] = request_data['enddate']
```

Comment:
Similar to my question above, why does this data need to be duplicated?

Reply:
in `export_all_rows_task` it expects `request_params` to have the date information in order to recreate the datespan object. When refactoring that code to be part of the __setstate__ like you mentioned in another comment, I could feasibly just have it check `request_data` if it can't find it in `request_params`.